### PR TITLE
[LPS-164582] As a developer, I want to retrieve my object log information from headless API

### PR DESCRIPTION
**Pull request by carolmariaabb**
**Related Issue**: https://issues.liferay.com/browse/LPS-179213

**Steps to reproduce**:

1. Enable the feature flag for [LPS-164582](https://issues.liferay.com/browse/LPS-164582)
3. Go to Control Panel -> Object -> Objects
4. Add a new Custom Object
5. Edit the Custom Object
6. In the details tab, go to Configuration -> Enable Entry History
7. In the details tab, go to Scope -> Panel Category Key -> Choose any option
8. Save the Custom Object
9. Go to fields tab -> Add any field
10. Publish the Custom Object
11. Go to Control Panel -> Configuration -> System Settings -> Security -> Audity -> Persistent Message Audit Message Processor -> Enable and Save
12. Add a new object entry in the custom object
13. Update this object entry as many times as you want
14. You can now see the object entry history log by doing:
```
curl -X 'GET' \
  'http://localhost:8080/o/c/customobjects/{objectEntryId}?nestedFields=auditEvents' \
  -H 'accept: application/json' \
  --user 'test@liferay.com:test'
```

**Obs.**: Make sure the user has `VIEW` and `ENTRY_HISTORY` permissions for the specific object entry

### CODE REVIEW

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Review by gabrielwas — CHANGES_REQUESTED** [review]
@carolmariaabb I don't see the implementation we need to do here that similar with the one we have for StructureContent versions.

There, we have multiple StructureContent versions attached to a StructureContent. But the "version" ones are just a extension of the StructureContent original one. So it does make sense to have this extension concept.

But here we're not versioning yet. We're just auditing. Exposing a totally different structure for it (AuditEvent). That way, the implementation should be different.

Let's chat about this.

File: modules/apps/object/object-rest-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -371,6 +371,25 @@ paths:
                             schema:
                                 $ref: "#/components/schemas/ObjectEntry"
             tags: ["ObjectEntry"]
+    "/by-external-reference-code/{externalReferenceCode}/audit-events":
+        get:
+            operationId: getByExternalReferenceCodeAuditEvent
```
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/object/object-rest-impl/rest-openapi.yaml:376]: @gabrielwas I agree but this name is always generated when I run buildRest and I can't change it

**Comment by gabrielwas** [comment]
Blocked: Waiting definition from PMs

**Comment by carolmariaabb** [comment]
PR updated to use `nestedFields` parameter to get object entry history.

/cc @gabrielwas

**Comment by carolmariaabb** [comment]
ci:test:sf

**Review by gabrielwas — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-rest-impl/rest-openapi.yaml [hunk 2]
```diff
@@ -1,5 +1,31 @@
...
+                        - $ref: "../../headless/headless-delivery/headless-delivery-impl/rest-openapi.yaml#Creator"
+                    readOnly: true
+                dateCreated:
+                    format: date
+                    readOnly: true
+                    type: string
+                eventType:
+                    readOnly: true
+                    type: string
+        AuditedFieldChange:
```
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/object/object-rest-impl/rest-openapi.yaml]: Done!

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Review by gabrielwas — APPROVED** [review]
LGTM! Thx @carolmariaabb !

**Comment by carolmariaabb** [comment]
ci:test:object

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
Forwarded without the integration test: https://github.com/brianchandotcom/liferay-portal/pull/132511

**Pull request closed without merging**